### ISSUE TITLE
Can not show banner ads on Android

### ISSUE THREAD

**Issue by zdd** [ratson/cordova-plugin-admob-free#338]
Use the latest version on my ionic project, but the banner ads not show on my Android device, here is the code: i call this method in ionViewDidLoad().
```
showBannerAds() {
    const bannerConfig: AdMobFreeBannerConfig = {
      id: 'ca-app-pub-2905597280622539/6707532901',
      isTesting: false,
      autoShow: true,
      overlap: false,
    };
    this.admobFree.banner.config(bannerConfig);
    console.log('start to prepare banner ads...');

    this.admobFree.banner.prepare()
      .then(() => {
        console.log('start to show banner ads...');
      })
      .catch(err => {
        console.log('show banner ads error: ', err);
      });
  }

```
and here is the error log from Android studio, just get 'Internal error', but what's the error really is? need more detail logs. any idea?

> 01-15 11:07:36.737 24251-24251/com.jiyuzhai.shufadaquan D/SystemWebChromeClient: file:///android_asset/www/build/56.js: Line 129 : start to show banner ads...
> 01-15 11:07:36.737 24251-24251/com.jiyuzhai.shufadaquan I/chromium: [INFO:CONSOLE(129)] "start to show banner ads...", source: file:///android_asset/www/build/56.js (129)
> 01-15 11:07:36.746 24251-24543/com.jiyuzhai.shufadaquan W/Ads: Update ad debug logging enablement as false
> 01-15 11:07:36.753 24251-24251/com.jiyuzhai.shufadaquan D/SystemWebChromeClient: file:///android_asset/www/build/56.js: Line 144 : start to show interstitial ads...
> 01-15 11:07:36.753 24251-24251/com.jiyuzhai.shufadaquan I/chromium: [INFO:CONSOLE(144)] "start to show interstitial ads...", source: file:///android_asset/www/build/56.js (144)
> 01-15 11:07:36.755 24251-24251/com.jiyuzhai.shufadaquan D/CordovaWebViewImpl: onPageFinished(file:///android_asset/www/index.html#/tabs/%E9%A6%96%E9%A0%81/home)
> 01-15 11:07:36.765 24251-24550/com.jiyuzhai.shufadaquan W/Ads: Update ad debug logging enablement as false
> 01-15 11:07:48.120 24251-24611/com.jiyuzhai.shufadaquan W/Ads: There was a problem getting an ad response. ErrorCode: 0
> 01-15 11:07:48.120 24251-24276/com.jiyuzhai.shufadaquan W/Ads: There was a problem getting an ad response. ErrorCode: 0
> 01-15 11:07:48.120 24251-24611/com.jiyuzhai.shufadaquan W/Ads: There was a problem getting an ad response. ErrorCode: 0
> 01-15 11:07:48.132 24251-24251/com.jiyuzhai.shufadaquan W/Ads: Failed to load ad: 0
> 01-15 11:07:48.136 24251-24251/com.jiyuzhai.shufadaquan D/CordovaWebViewImpl: >>> loadUrl(javascript:cordova.fireDocumentEvent('admob.interstitial.events.LOAD_FAIL',{"error":0,"reason":"Internal error","adType":"interstitial"});)
> 01-15 11:07:48.139 24251-24251/com.jiyuzhai.shufadaquan D/CordovaWebViewImpl: >>> loadUrl(javascript:cordova.fireDocumentEvent('onFailedToReceiveAd',{"error":0,"reason":"Internal error","adType":"interstitial"});)
> 01-15 11:07:48.142 24251-24251/com.jiyuzhai.shufadaquan W/Ads: Failed to load ad: 0
> 01-15 11:07:53.903 24251-24251/com.jiyuzhai.shufadaquan D/CordovaWebViewImpl: >>> loadUrl(javascript:cordova.fireDocumentEvent('admob.banner.events.LOAD_FAIL',{"error":0,"reason":"Internal error","adType":"banner"});)
> 01-15 11:07:53.907 24251-24251/com.jiyuzhai.shufadaquan D/CordovaWebViewImpl: >>> loadUrl(javascript:cordova.fireDocumentEvent('onFailedToReceiveAd',{"error":0,"reason":"Internal error","adType":"banner"});)

**Comment by edento** [comment]
I are you sure you're having an error? because it is not even logged. it seems you have forgotten to call `this.admobFree.banner.show();` method.

```
showBannerAds() {
    const bannerConfig: AdMobFreeBannerConfig = {
      id: 'ca-app-pub-2905597280622539/6707532901',
      isTesting: false,
      autoShow: true,
      overlap: false,
    };
    this.admobFree.banner.config(bannerConfig);
    console.log('start to prepare banner ads...');

    this.admobFree.banner.prepare()
      .then(() => {
        console.log('start to show banner ads...');
         this.admobFree.banner.show(); // <<<<< here
      })
      .catch(err => {
        console.log('show banner ads error: ', err);
      });
  }
```

I hope this solves your issue.

**Comment by zdd** [comment]
> I are you sure you're having an error? because it is not even logged. it seems you have forgotten to call `this.admobFree.banner.show();` method.
> 
> ```
> showBannerAds() {
>     const bannerConfig: AdMobFreeBannerConfig = {
>       id: 'ca-app-pub-2905597280622539/6707532901',
>       isTesting: false,
>       autoShow: true,
>       overlap: false,
>     };
>     this.admobFree.banner.config(bannerConfig);
>     console.log('start to prepare banner ads...');
> 
>     this.admobFree.banner.prepare()
>       .then(() => {
>         console.log('start to show banner ads...');
>          this.admobFree.banner.show(); // <<<<< here
>       })
>       .catch(err => {
>         console.log('show banner ads error: ', err);
>       });
>   }
> ```
> I hope this solves your issue.

No, that's not the case, I have set autoShow: true, it will automatically show ads.

**Comment by AhSem** [comment]
I am having the same problem as well. In iOS, the ads is showing but it is written "Test Ads", but in Android, it is not showing despite i put autoShow: true, and also called the function to show banner ad explicitly.

**Comment by PELOX** [comment]
Hi Folks!
Having similar issue with android platform and ionic. I have been fighting for several days and can not find the way to do ir work  on android.

My experience.
IOS: Seemes that do not use the app id  and works with android app id both values for isTesting.

Android: Can not avoid the black box with banner and no intersticial is presented.  In both cases no errors appear in the log. Do not works in each of the isTenting values ;-(
In case of android i have tested with virtual machine and in real device (3 devices)

Adbmob account seemes to be ok and, by now i can check that  devices are doing some request but no presentation are done.

If some one is experiencing same issue, posible we can share a solution or workarround

Best regards

**Comment by PELOX** [comment]
Update: i was able to catch the return of admob platform, **Error Code:3 Description: "No Fill"** but still blocked with the issue :-(

**Comment by Mathewkutty** [comment]
> Update: i was able to catch the return of admob platform, **Error Code:3 Description: "No Fill"** but still blocked with the issue :-(

This error mainly because Google AdMob doesn't have any ads to respond to your request.You can  use mediation group for handling this ad shortage issue
